tools/set_user_role.py

```diff
@@ -0,0 +1,116 @@
+GREEN = '\033[92m'
+RED = '\033[91m'
+YELLOW = '\033[93m'
+
+
+def get_users(role=None):
+    return [user[0] for user in DBSession().execute(sa.select(User))]
+
+
+def roles_user_has(user):
+    return [role.id for role in user.roles]
+
+
+def roles_user_does_not_have(user):
+    return [role for role in role_acls if role not in roles_user_has(user)]
+
+
+def list_users():
+    users = get_users()
+    if len(users) == 0:
+        print('\nNo users in database')
+    else:
+        # print each user's username and roles
+        print(f'\n{BOLD}List of users and current roles:{END}')
+        for i, user in enumerate(users):
+            list_of_new_roles = (
+                BOLD
+                + GREEN
+                + f'{END}, {BOLD}{GREEN}'.join(roles_user_does_not_have(user))
+                + END
+            )
+            if len(user.roles) == 0:
+                print(
+                    f'\n{BOLD}{YELLOW}{user.username}{END} (no roles) can be elevated to {list_of_new_roles}'
+                )
+            else:
+                list_of_roles = (
+                    BOLD + RED + f'{END}, {BOLD}{RED}'.join(roles_user_has(user)) + END
+                )
+                print(
+                    f'\n{BOLD}{i+1}. {YELLOW}{user.username}{END} has the following roles: {list_of_roles} and can be elevated to: {list_of_new_roles}'
+                )
+        print('\n')
+
+
+def set_user_role(username=None, role=None):
+    if not role:
+        print(
+            f'{BOLD}{RED}\nNo role provided;{END} setting to {BOLD}{GREEN}Super admin{END}{BOLD}{END}.'
+        )
+        role = 'Super admin'
+    if role not in role_acls:
+        print(f'{BOLD}{RED}\nRole not found!{END} Try a role from the list below:\n')
+        for i, role in enumerate(role_acls):
+            print(f'{BOLD}{i+1}. {GREEN}{role}{END}')
+        print('\n')
+
+    elif username is not None:
+        users = get_users()
+        if username in [user.username for user in users]:
+            if role not in [
+                role.id
+                for role in [
+                    user.roles[0] for user in users if user.username == username
+                ]
+            ]:
+                setup_permissions()
+                add_user(username, roles=[role], auth=True)
+                print(
+                    f'\nSuccessfully assigned role {BOLD}{GREEN}{role}{END} to {BOLD}{YELLOW}User {username}{END}\n'
+                )
+            else:
+                print(
+                    f'\nUser {BOLD}{YELLOW}{username}{END} already has role {BOLD}{GREEN}{role}{END}\n'
+                )
+
+        else:
+            print(
+                f'\n{BOLD}{RED}User{END} {BOLD}{YELLOW}{username}{END} {BOLD}{RED}does not exist{END}\n'
+            )
+
+
+def main():
+    if args.list or args.username:
+        if args.list:
+            list_users()
+        if args.username:
+            set_user_role(args.username, args.role)
+    else:
+        print(
```

Comment:
I thought about this more, and I like your original version: just print the help string here and don't try and connect to the db.